scene/gui/text_edit.cpp

```diff
@@ -7818,7 +7818,10 @@ void TextEdit::_update_selection_mode_word(bool p_initial) {
 		int origin_col = is_new_selection_dir_right ? carets[caret_index].selection.word_begin_column : carets[caret_index].selection.word_end_column;
 		int caret_col = is_new_selection_dir_right ? end : beg;
 
-		select(origin_line, origin_col, line, caret_col, caret_index);
+		// Expand the word selection only if the caret is not at the start of the selection.
+		if (column != carets[caret_index].selection.word_begin_column || line != origin_line) {
+			select(origin_line, origin_col, line, caret_col, caret_index);
+		}
```

Comment:
```suggestion
		if (column != carets[caret_index].selection.word_begin_column || line != origin_line || carets[caret_index].selection.word_begin_column == carets[caret_index].selection.word_end_column) {
			select(origin_line, origin_col, line, caret_col, caret_index);
		}
```

This prevents text from deselecting even if there was no selection to start with. I think adding this condition will fix it.

Reply:
Thanks! I added the tests (hopefully) as you described. This condition allows it to pass the new test.